packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,241 @@
+use crate::types::DbPool;
+
+use sqlx::postgres::PgRow;
+use sqlx::types::chrono;
+
+use crate::models::user::User;
+
+use sqlx::types::Uuid;
+use sqlx::Row;
+
+pub struct Group {
+    pub id: i32,
+    pub uuid: Uuid,
+    pub group_name: String,
+    pub created: chrono::NaiveDateTime,
+    pub members: Vec<Uuid>,
+}
+
+pub struct UserGroup {
+    pub id: i32,
+    pub user_id: i32,
+    pub group_id: i32,
+    pub role_id: Role,
+    pub created: chrono::NaiveDateTime,
+}
+
+#[derive(Debug, PartialEq)]
+#[repr(u8)]
+pub enum Role {
+    Admin = 0,
+    Member = 1,
+}
+
+impl TryFrom<u8> for Role {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Role::Admin),
+            1 => Ok(Role::Member),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(sqlx::FromRow)]
+pub struct UserIDs {
+    pub user_ids: Vec<String>,
+}
+
+#[derive(sqlx::FromRow)]
+pub struct GroupIDs {
+    pub group_ids: Vec<UserGroup>,
+}
+
+pub async fn delete_user_groups(db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+    sqlx::query("DELETE FROM user_group WHERE user_id = $1")
+        .bind(user_id)
+        .execute(db)
+        .await?;
+    Ok(())
+}
+
+pub async fn get_user_groups(db: &DbPool, user_id_in: i32) -> Result<Vec<UserGroup>, sqlx::Error> {
+    let mut user_groups: Vec<UserGroup> = vec![];
```

Comment:
this needs to be improved

Reply:
i could not get map and closures to work, we verbally agreed that loops were fine

Reply:
also this is not used at the present, gannon requested and it MIGHT be used in future

Reply:
removed